dev/com.ibm.ws.com.unboundid/src/com/ibm/ws/com/unboundid/InMemoryLDAPServer.java

```diff
@@ -47,37 +57,66 @@
      * service.
      *
      * @param bases The base entries to create for this in-memory LDAP server.
      * @throws Exception If something went wrong
      */
-    public InMemoryLDAPServer(String... bases) throws Exception {
+    public InMemoryLDAPServer(boolean useWimSchema, boolean useSSLConfig, String... bases) throws Exception {
 
         config = new InMemoryDirectoryServerConfig(bases);
         config.addAdditionalBindCredentials(getBindDN(), getBindPassword());
-        config.setListenerConfigs(
-                                  InMemoryListenerConfig.createLDAPConfig("LDAP", // Listener name
-                                                                          null, // Listen address. (null = listen on all interfaces)
-                                                                          0, // Listen port (0 = automatically choose an available port)
-                                                                          null) // StartTLS factory
-        ); // Client factory
 
+        if (useSSLConfig) {
+            keystore = extractResourceToFile("/resources/keystore.jks", "keystore", ".jks").getAbsolutePath();
+            final SSLUtil serverSSLUtil = new SSLUtil(new KeyStoreKeyManager(keystore, keystorePassword
+                            .toCharArray(), "JKS", "cert-alias"), new TrustAllTrustManager());
+            config.setListenerConfigs(InMemoryListenerConfig.createLDAPSConfig("LDAPS", 0, serverSSLUtil.createSSLServerSocketFactory()));
+        } else {
+            config.setListenerConfigs(InMemoryListenerConfig.createLDAPConfig("LDAP", null, 0, null));
+        }
+        Schema schema = null;
+        if (useWimSchema) {
+            InputStream in = getClass().getResourceAsStream("/resources/wimschema.ldif");
+            Schema wimschema = Schema.getSchema(in);
+            schema = Schema.mergeSchemas(Schema.getDefaultStandardSchema(), wimschema);
+        }
+        config.setSchema(schema);
+
+        ds = new InMemoryDirectoryServer(config);
+        ds.startListening();
+    }
+
+    /**
+     * Extract a resource from the JAR to a temporary file on the file system. The
+     * file is marked for deletion on JVM exit.
+     *
+     * @param resource The resource from the JAR to extract.
+     * @param prefix   Prefix for the temporary file.
+     * @param suffix   Suffix for the temporary file.
+     * @return The {@link File} instance.
+     * @throws IOException if there was an issue extracting the file.
+     */
+    protected File extractResourceToFile(String resource, String prefix, String suffix) throws IOException {
+        File tempfile = File.createTempFile(prefix, suffix);
+        InputStream src = getClass().getResourceAsStream(resource);
+        Files.copy(src, Paths.get(tempfile.getAbsolutePath()), StandardCopyOption.REPLACE_EXISTING);
+        return tempfile;
+    }
+
+    public InMemoryLDAPServer(String... bases) throws Exception {
```

Comment:
Missing Javadoc. Move the other constructor's javadoc down here.

Move this in front or right after the other constructor.